python/hackathon_app/sheets.py

```diff
@@ -54,20 +63,22 @@ def register_user(self, hackathon: str, user: "User"):
             self.registrations.register(registrant)
 
 
-TStructure = Union[Sequence["Hackathon"], Sequence["User"], Sequence["Registrant"]]
+TModel = TypeVar("TModel")
 
 
-class WhollySheet:
-    def __init__(self, client, spreadsheet_id: str, sheet_name: str, structure):
+class WhollySheet(Generic[TModel]):
```

Comment:
here's the interesting part of this PR: by subclassing `Generic[TModel]` we can use the `TModel` generic as our type inside the class and then subclassers will say exactly which real type to use e.g.

```python
class Users(WhollySheet[User])
```